### ISSUE TITLE
Kreya crashes suddenly without any reproducibility in Mac M1

### ISSUE THREAD

**Issue by harshvladha** [riok/Kreya#97]
**Describe the bug**
Kreya crashes suddenly without any reproducibility in Mac M1

Earlier the app used to crash frequently, the number of crashes has reduced definitely but has not gone to zero.

**To Reproduce**
No way to reproduce

**Expected behavior**
Do not crash

**Screenshots**
Typical mac os dialog on a crash which has a button to: report, ignore and relaunch.

will add a screenshot when it crashes again.

**Environment (if possible, copy the information from the error dialog or the About menu):**
 - OS: Mac OS Monterey 12.4
 - Kreya Version: 1.8.0

**Additional context**
- Using gRPC APIs
- API responses are not malformed
- API request doesn't change a lot
- No consistent pattern for crashing

**Comment by latonz** [comment]
Thanks for your bug report! Could you add the contents of the crash report and logs (`~/.config/Kreya/kreya.log`) to the issue (you may need to check for personal details before uploading...)?

**Comment by harshvladha** [comment]
@latonz - `~/.config/Kreya/kreya.log` doesn't have the logs for the time when Kreya crashed suddenly. I only see logs of date before creating this issue or the one which came later but is not related to this.





```
2022-06-28 08:44:56.243 +05:30 [ERR] unhandled exception occured
System.NullReferenceException: Object reference not set to an instance of an object.
2022-06-29 10:44:32.781 +05:30 [ERR] unhandled exception occured
System.NullReferenceException: Object reference not set to an instance of an object.
2022-07-03 11:02:06.904 +05:30 [ERR] unhandled exception occured
System.NullReferenceException: Object reference not set to an instance of an object.
2022-07-17 21:59:08.631 +05:30 [ERR] unhandled exception occured
System.NullReferenceException: Object reference not set to an instance of an object.
2022-07-18 12:48:25.645 +05:30 [ERR] unhandled exception occured
2022-07-21 09:12:08.270 +05:30 [ERR] unhandled exception occured
System.NullReferenceException: Object reference not set to an instance of an object.
2022-07-27 23:32:55.203 +05:30 [ERR] exception while trying to fetch updates
System.Net.Http.HttpRequestException: No route to host (stable-downloads.kreya.app:443)
 ---> System.Net.Sockets.SocketException (65): No route to host
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationTo$
   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)
   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationTo$
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, Canc$
   at System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsyncCore[T](Task`1 taskResponse, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Kreya.Core.Updater.AutoUpdaterService.FindUpdate(CancellationToken ct)
   at Kreya.Core.Updater.AutoUpdaterService.CheckForUpdates(CancellationToken ct)
2022-07-28 02:47:01.929 +05:30 [ERR] unhandled exception occured
System.NullReferenceException: Object reference not set to an instance of an object.




```

**Comment by harshvladha** [comment]
I will update the thread once kreya crashes and mac reporter dialog comes.

**Comment by senthilkumarv** [comment]
[Kreya.app-2022-08-02-160114.txt](https://github.com/riok/Kreya/files/9245280/Kreya.app-2022-08-02-160114.txt)
Here is a log. I consistently crashes with bad access or seg fault

**Comment by harshvladha** [comment]
```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               Kreya.app [97881]
Path:                  /Applications/Kreya.app/Contents/MacOS/Kreya.app
Identifier:            app.kreya
Version:               1.8.0 (1.8.0)
Code Type:             X86-64 (Translated)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2022-08-03 09:17:29.4265 +0530
OS Version:            macOS 12.4 (21F79)
Report Version:        12
Anonymous UUID:        6D13833E-EBA6-8DE9-AD6B-64C27944F1D9

Sleep/Wake UUID:       775A347A-C9F8-49C3-80F4-7119AA5BE0B2

Time Awake Since Boot: 730000 seconds
Time Since Wake:       3136 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called


Kernel Triage:
VM - Compressor failed a blocking pager_get
VM - Compressor failed a blocking pager_get
VM - Compressor failed a blocking pager_get
VM - Compressor failed a blocking pager_get
VM - Compressor failed a blocking pager_get

```


Mac OS Crash report

**Comment by latonz** [comment]
Thank you for the crash logs.
Could it possible be that you are running the x86 version of Kreya instead of the arm version?
To verify this run `file /Applications/Kreya.app/Contents/MacOS/Kreya.app`. You may remove the x64 version and download the arm64 version from https://kreya.app/downloads/

**Comment by harshvladha** [comment]
```
>> file /Applications/Kreya.app/Contents/MacOS/Kreya.app                                                                                          ✔ 
/Applications/Kreya.app/Contents/MacOS/Kreya.app: Mach-O 64-bit executable x86_64
```

**Comment by harshvladha** [comment]
After using ARM version for a while, it seems to be crashing again


<img width="985" alt="Screenshot 2022-08-19 at 9 27 11 AM" src="https://user-images.githubusercontent.com/7391816/185539785-4c2b03f4-1275-4cc0-bd5c-74b58a6b59a4.png">

**Comment by CommonGuy** [comment]
This should be fixed with the newly released 1.9 release, which contains bugfixes for macOS.
